### PR TITLE
Added function to PulseBase to calculate spectrogram

### PR DESCRIPTION
**Pull request by DanHickstein**
[Dudley](http://journals.aps.org/rmp/abstract/10.1103/RevModPhys.78.1135) makes a big deal about calculating the spectrogram of the pulse (for example, see his Fig. 10). I included a function in PulseBase to allow the spectrogram to be calculated.

The results look something like this (pulse after propagation in a nonlinear fiber):

figure_4

Note: this PR should be merged after (or in place of) https://github.com/ycasg/PyNLO/pull/14.

### CODE REVIEW

File: src/pynlo/media/fibers/fiber.py [hunk 1]
```diff
@@ -67,7 +67,36 @@ def load_from_db(self, length, fibertype, poly_order = 2):
...
+        This loads dispersion give the path of a file. 
+        The file is expected to be in the format
+        wavelength (nm), D (ps/nm/km).
+        """
+        import os
+        
+        if fiberName == None:
+            self.fibertype = os.path.basename(filename)
+        else:
+            self.sibertype = fiberName
```
Comment (ycasg) [inline src/pynlo/media/fibers/fiber.py]: Typo

**Comment by ycasg** [comment]
This looks really nice. The spectrogram calculation is a really nice addition.

**Comment by DanHickstein** [comment]
Typo? You don't use nonlinear optical sibers?

**Comment by DanHickstein** [comment]
Fixed the typo. Can this be merged now?

**Pull request merged**